### ISSUE TITLE
RSS feeds as external actors?

### ISSUE THREAD

**Issue by alxlg** [tootsuite/mastodon#10134]
At the moment there are many RSS -> Mastodon bots in the Fediverse.

Maybe instead of using bots could we subscribe to RSS feeds from Mastodon?

I imagine the following UX:
- Paste an RSS feed URL in Mastodon search 
- Mastodon shows a profile filled with RSS feed info and that has feed entries as toots (maybe title + link)
- Subscribe to that profile like any other Mastodon/ActivityPub actor 
- Enjoy the RSS feed entries in the timeline 
- Share/favourite/comment those entries like normal toots
- Mastodon should serve RSS-based toots and profiles to Mastodon clients as normal toots/profiles 

Let's discuss if this is viable/desiderable...

**Comment by wakest** [comment]
This would lesson the amount of bots that people keep needing to make to follow the content they want to follow. I think the issue mainly would be when a post is shared, who's account would it come from? would we thus be having to create a mini bot anyway everytime a rss feed is added? and how would this interact with the fediverse outside of mastodon's context.

**Comment by Xanthec** [comment]
You can have a look how Hubzilla already implemented that feature.

**Comment by trwnh** [comment]
Not sure this makes 100% sense. You couldn't reply to an RSS entry in a consistent way, "following" an RSS feed is not entirely the same as following an actual person, and I think there are already several services that convert RSS into ActivityPub mirrors. You could ask for being able to subscribe to RSS feeds within a column, but that would make Mastodon a multiprotocol application and dilute its use case as an explicitly social server. What you're asking for would be better-served by a feed reader application.

**Comment by alxlg** [comment]
@trwnh 
> You couldn't reply to an RSS entry in a consistent way 

Why?

> "following" an RSS feed is not entirely the same as following an actual person 

Why? Because Mastodon server would fetch the RSS feed? What's wrong with it?

> I think there are already several services that convert RSS into ActivityPub mirrors 

As I said I wonder if running bots is the most convenient way

> but that would make Mastodon a multiprotocol application and dilute its use case as an explicitly social server 

There are already a lot of RSS -> Mastodon bots so this is definetly an use case and it's already there, I'm just asking if using bots is the most convenient way 

> What you're asking for would be better-served by a feed reader application 

... really? :stuck_out_tongue_winking_eye:

**Comment by jalcine** [comment]
Yeah, to be frank; this is definitely outside of the purview of what Mastodon would be. A [social reader](http://indieweb.org/reader) that'd let you view your Mastodon timelines but also your preferred feeds would be optimal (as well as letting you reply using your preferred platform).

**Comment by alxlg** [comment]
@jalcine 
> A social reader that'd let you view your Mastodon timelines but also your preferred feeds would be optimal 

We already have RSS feed readers and each Mastodon profile have an RSS feed so one can already just use an RSS reader.

But people run RSS -> Mastodon bots, this is the point. Why they do so? Probably because it's convenient to do so:

- you can mix real profiles with RSS bots in your list of subscribtion that is stored online in Mastodon and shared across Mastodon clients 
- feed entries appears as toots so you can just click a button to boost/like consistently with real toots 
- if a contact of yours boosted a toot by an RSS bot and you boost it too you contribute to increase the boost counter, the toot will appear in your followers' timeline and they see the OP is RSS bot, not your contact 
- if a contact of yours boost a toot by an RSS bot you can easly open the RSS bot profile and subscribe to it to receive future feed entries as toots 

How is this out of scope for Mastodon? I can't believe Mastodon and other (federated and centralized) social networks don't integrate RSS feeds. Also there should be an ActivityPub spec on how to integrate RSS feeds in the federation.

**Comment by hermann-san** [comment]
The only bot I know is feet2toot, mention in #11587 . Are there more?

**Comment by mancoi** [comment]
Hello, I’m a Mastodon user searching for a way to read RSS feeds on Mastodon and found this feature request, but it seems like this request is currently being overlooked/underestimated.

If a Mastodon instance can fetch and show the RSS items on a user’s RSS timeline, it would be a killer feature. I mean, everyone would be able to build a ‘tweetdeck’ for themselves to catch up on their interested news, live events, what’s trending, etc.

Because Mastodon doesn’t have an algorithm, RSS is the way for users to build their own timeline while still having the ability to do social stuff. If you could somehow allow users to create and share multiple RSS timelines and others could follow those RSS timelines, I think it would be very convenient, especially for newly joined users. One-click and now you follow all the news sites talking about your hobbies, and you can customize it however you want, isn't that nice?

Another use case is that some people want/need to have a convenient way to toot about something they just read without doing copy-paste. For example, you read news in a foreign language → translate it → toot with the source included (Yes sir, being able to quote toot is also a nice-to-have feature).

I just wanted to share some of my thoughts and hope Mastodon will be able to handle RSS in the future. It’s a real need, but it’s up to you to decide if Mastodon needs this feature or not.

**Comment by markcellus** [comment]
OpenRSS.org provides feeds for Mastodon. I've been using that until this is implemented.

https://openrss.org/blog/mastodon-rss-feeds

**Comment by mancoi** [comment]
> OpenRSS.org provides feeds for Mastodon. I've been using that until this is implemented.
> 
> https://openrss.org/blog/mastodon-rss-feeds

uhm, I think this issue is about to read RSS on Mastodon, not getting Mastodon RSS feed

**Comment by markcellus** [comment]
Oops sorry, misinterpreted will delete

**Comment by LinkCodeDev** [comment]
For those who don't know how it would work: Friendica already implements this feature.
This is THE feature I'm most looking forward to: 90% of what I do online will be possible from Mastodon, and I won't need to use multiple websites anymore.
In practical terms, every 12 hours (configurable), the instance would check if the RSS feed has changed and display the update.
